modules/twinklewarn.js

```diff
@@ -102,13 +102,37 @@ Twinkle.warn.callback = function twinklewarnCallback() {
 		type: 'div',
 		label: '',
 		style: 'color: red',
-		id: 'twinkle-warn-olddiff-message'
+		id: 'twinkle-warn-revert-messages'
 	});
 
-	// Confirm edit wasn't too old for a warning
 	var vanrevid = mw.util.getParamValue('vanarticlerevid');
 	if (vanrevid) {
-		var query = {
+		var message = '';
+		var query = {};
+
+		// If you tried reverting, check if *you* actually reverted
+		if (!mw.util.getParamValue('noautowarn') && mw.util.getParamValue('vanarticle')) { // Via fluff link
+			query = {
+				action: 'query',
+				titles: mw.util.getParamValue('vanarticle'),
+				prop: 'revisions',
+				rvstartid: vanrevid,
+				rvlimit: 2,
+				rvdir: 'newer',
+				rvprop: 'user'
+			};
+
+			new Morebits.wiki.api('Checking if you successfully reverted the page', query, function(apiobj) {
+				var revertUser = $(apiobj.getResponse()).find('revisions rev')[1].getAttribute('user');
+				if (revertUser && revertUser !== mw.config.get('wgUserName')) {
+					message += ' Someone else besides you actually reverted the page, and may have already warned them.';
```

Comment:
Similar to Huggle, suggest changing this to:
"Someone else has already reverted the page and may have warned the user."

Reply:
Combined the two: "Someone else reverted the page and may have already warned the user"